src/containers/List/ListWrapper.tsx

```diff
@@ -105,7 +108,17 @@ const ListWrapper: React.FC = () => {
     }
   }
 
-  if (loading ) return <Loading />
+  if (loading) return <Loading />
+
+  let filterDefinitions: FilterDefinitions = {}   
+  
+  columns.forEach((column) => {
+    const filter = Object.entries(APPLICATION_FILTERS).find(([filterName, _]) => camelCase(column.sortName) === filterName)
```

Comment:
Yeah, I think the problem is that you've changed the "sortName" for some of the columns which has messed up how the list looks/works.

And it would be better if this logic was hidden away in the hook it relates to rather than here. You might be able to run the function to check whether filter should be visible in the `useApplicationFilters` hook itself.

So each filterDefinition could be something like:
```
  applicantDeadline: {
      type: 'date',
      default: defaultFilters.some((filter) => filter === 'applicantDeadline'),
      visible: checkFilterVisibility("DEADLINE_DATE")
      title: strings.FILTER_APPLICANT_DEADLINE,
      options: { namedDates: NAMED_DATE_RANGES },
    },
```

And then create a function `checkFilterVisibility` to do the logic you've done here, ie. compare against column visibility. And then you don't need to match by "sortName" (which causes the problems I described elsewhere), you can just directly pass the column to compare against to "checkFilterVisibility" for each filter.

Reply:
Thanks Carl, I didn't think this was a very good solution either. I was trying to make it without many changes... I still don't get one thing though... I'll check with you before starting on this

Reply:
Ok! Quite a lot of changes... sorry! But the result of what you and I talked is there. Hopefully much more like what was envisioned at the start : )